src/formatting-model-interfaces.d.ts

```diff
@@ -0,0 +1,593 @@
+            DisplayNameUpdated = "DisplayNameUpdated",
+        }
+
+        interface ListEditor extends SimpleComponentBase<ListEditorValue> { }
+
+        interface ImageUpload extends SimpleComponentBase<ImageValue> { }
+
+        interface DurationPicker extends SimpleComponentBase<string> {
+            /** These are specfic to this control */
+            validators?: {
+                min?: string;
+                max?: string;
+                integer?: boolean;
+            };
+        }
+
+        /** This will be used for showing only "fx" button without any other content */
+        interface EmptyControl extends SimpleComponentBase<void> { }
+
+        interface GradientBar extends SimpleComponentBase<string> { }
+
+        interface ShapeMapSelector extends SimpleComponentBase<GeoJson> {
+            isAzMapReferenceSelector?: boolean;
+        }
+
+        interface DatePicker extends SimpleComponentBase<Date> {
+            placeholder: string;
+            validators?: {
+                max?: MaxValidator<Date>;
+                min?: MinValidator<Date>;
+            };
+        }
+
+        interface ReadOnlyText extends SimpleComponentBase<string> { }
+
+        interface ConditionalFormattingControl extends SimpleComponentBase<powerbi.VisualObjectRepetition.VisualObjectRepetitionMetadata> {
+            displayName: string;
+        }
+
+        interface Link extends SimpleComponentBase<() => (any | void)> {
+            ariaLabel?: string;
+        }
+
+        interface SeriesDialogLink extends Link {
+            type: FormattingComponent.SeriesDialogLink;
+            warningIcon?: boolean;
+            resetToDefaultDescriptors?: FormattingDescriptor[];
+        }
+    }
+}
+
+declare namespace powerbi {
+    export interface ThemeColorData {
+        value: string;
+        id?: number;
+        shade?: number;
+    }
+}
+
+declare namespace powerbi {
+    export interface ValidationRequiredInfo {
+        required?: boolean;
+    }
+
+    export interface ValidationNumericInfo {
+        decimal?: boolean;
+        integer?: boolean;
+        min?: number | Date | string;
+        max?: number | Date | string;
+    }
+}
+
+declare namespace powerbi.explore.directives {
+    type ValidationAllInfo = powerbi.ValidationRequiredInfo & powerbi.ValidationNumericInfo;
+    export interface ValidationInfo extends ValidationAllInfo {
+        url?: boolean;
+        field?: FieldValidationInfo;
+    }
+
+    export interface FieldValidationInfo {
+        kind: FieldKind;
+        // For more than 1 type to validate on, we can use Variant type descriptor
+        type?: ValueTypeDescriptor;
+    }
+}
+
+declare module powerbi {
+    export interface IVisualErrorMessage {
+        message: string;
+        title: string;
+        detail: string;
+    }
+    export interface IVisualWarning {
+        code: string;
+        // getMessages(resourceProvider: jsCommon.IStringResourceProvider): IVisualErrorMessage;
```

Comment:
needed?

Reply:
If you mean the commented method so no, I can remove it